tiny-event-sourcing-lib/src/main/kotlin/ru/quipy/streams/EventStoreReader.kt

```diff
@@ -0,0 +1,142 @@
+package ru.quipy.streams
+
+import kotlinx.coroutines.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import ru.quipy.database.EventStore
+import ru.quipy.domain.EventRecord
+import ru.quipy.domain.EventStreamReadIndex
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+interface EventReader {
+    suspend fun read(batchSize: Int): List<EventRecord>
+    fun postProcessRecord(eventRecord: EventRecord)
+
+    /**
+     * We can "replay" events in the stream by resetting it to desired reading index
+     */
+    fun resetReadIndex(resetInfo: ReadIndexResetInfo)
+    fun stop()
+}
+
+class EventStoreReader(
+    private val eventStore: EventStore,
+    private val streamName: String,
+    private val tableName: String,
+    private val streamManager: EventStreamReaderManager,
+    private val eventStreamNotifier: EventStreamNotifier,
+    private val dispatcher: CoroutineDispatcher
+) : EventReader {
+    companion object {
+        private val NO_RESET_REQUIRED = ReadIndexResetInfo(-1)
+    }
+
+    private val logger: Logger = LoggerFactory.getLogger(EventStoreReader::class.java)
+    private val nextReaderAliveCheck: Duration = 15.seconds
+
+    private val isReaderPrimary: AtomicBoolean = AtomicBoolean(false)
+    private val isScanActive: Boolean = true
+
+    private val scanJob: Job = launchEventStoreReaderScanJob()
+
+    private var streamReadIndex: EventStreamReadIndex = EventStreamReadIndex(streamName, readIndex = 0L, version = 0L)
```

Comment:
the class is called "reader", there is no notion of the "stream" here, maybe rename?

Reply:
Didn't catch whether you want to rename type or variable name?